Features/update for tsc spec compliance

AustEcon opened the pull request:
As discussed with @jadwahab and @sirdeggen here is the PR for updating to the TSC spec: https://tsc.bitcoinassociation.net/standards/direct_payment_protocol/

Corresponding dual PR here: https://github.com/libsv/go-dpp/pull/69

So far this only updates the `PaymentRequest` structure which is enough for now to get ElectrumSV through the first part of the data flow for development...

Still to do:

- [x] `Payment`
- [x] `PaymentACK`
- [x] Adequate testing end-to-end against this code

Until the `dpp-proxy` is updated to the TSC spec, ElectrumSV will be building against this feature branch to avoid technical debt of the old message structures.

AustEcon committed:
Pinned github.com/libsv/go-dpp to a specific commit on feature branch: `features/updates-for-tsc-spec-compliance`

- This in turn results in updates of the other vendored dependencies

AustEcon committed:
Update to latest TSC DPP spec. See: https://tsc.bitcoinassociation.net/standards/direct_payment_protocol/#HybridPaymentMode(BRFCID:ef63d9775da5)

jadwahab commented on data/noop/noop.go:
```diff
@@ -34,29 +32,43 @@ func (n *noop) PaymentCreate(ctx context.Context, args dpp.PaymentCreateArgs, re
 func (n noop) PaymentRequest(ctx context.Context, args dpp.PaymentRequestArgs) (*dpp.PaymentRequest, error) {
```
shouldn't this be called `PaymentTerms` actually?

AustEcon replied:
Yes. You are right. I will change it

AustEcon committed:
Update to latest TSC DPP spec. continued... now includes latest `Payment` and `PaymentACK`

AustEcon committed:
Refactoring to rename `PaymentRequest` to `PaymentTerms` and updating to the latest version of `go-dpp` feature branch commit

rt121212121 committed:
Secure payment term support (relies on existing go-dpp changes) (#60)

AustEcon committed:
Silence false positive golangci-lint warning for duplicate code block

AustEcon committed:
Pin to latest commit on go-dpp feature branch: ` features/updates-for-tsc-spec-compliance`

AustEcon committed:
Make error response more readable

rt121212121 committed:
Update `dpp-proxy` so that the default method of hosting invoices is the `JSONEnvelope` based signed model. This also removes `payd` as a dependency, although there are remnants of configuration variables.

rt121212121 committed:
Update the `go-dpp` version for the CI passing tag.

rt121212121 committed:
Pin `swag` to 1.8.4 which is the newest release that does not 'panic: runtime error: invalid memory address or nil pointer dereference'. Generate the swagger update.

AustEcon committed:
Return errors back to payer as a 400 bad request http error

- This includes relayed `payment.error` messages over the websocket from the payee (for example: `bad-txns-premature-spend-of-coinbase`)

AustEcon committed:
Updates to error handling including removal of the `Memo` and `Error` fields of the dpp.PaymentACK struct (as this is not a part of the TSC specification)

- The payee wallet returns a `server.ClientError` in the body of the `payment.error` websocket message (this has not changed) and this is parsed and translated into the appropriate http error response. The `PaymentACK` was being inappropriately used in some cases as a way to pass around error details. This is no longer the case.
- A `PaymentACK` object is only returned in the cased of successful payment.

AustEcon committed:
Update to latest `go-dpp` version

AustEcon committed:
Rebase on master and remove `vendor` folder
